core/src/style/textStyle.cpp

```diff
@@ -98,14 +102,36 @@ struct TextStyleBuilder : public StyleBuilder {
 
     bool prepareLabel(const TextStyle::Parameters& _params, Label::Type _type);
     void addLabel(const TextStyle::Parameters& _params, Label::Type _type, Label::Transform _transform);
+
+    std::string resolveTextSource(const std::string& textSource, const Properties& props) const;
 };
 
 bool TextStyleBuilder::checkRule(const DrawRule& _rule) const {
     return true;
 }
 
+std::string TextStyleBuilder::resolveTextSource(const std::string& textSource, const Properties& props) const {
+
+    std::vector<std::string> fallbacks;
+    std::stringstream ss(textSource);
+    std::string item;
+
+    // Parse fallbacks
+    while (std::getline(ss, item, ',')) {
+        fallbacks.push_back(item);
```

Comment:
Can we just check and return the property here? Do we need that `fallbacks` vector?